packages/strapi-plugin-users-permissions/services/User.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+/**
+ * User.js service
+ *
+ * @description: A set of functions similar to controller's actions to avoid code duplication.
+ */
+
+// Public dependencies.
+const _ = require('lodash');
+const bcrypt = require('bcryptjs');
+
+module.exports = {
+
+  /**
+   * Promise to fetch all users.
+   *
+   * @return {Promise}
+   */
+
+  fetchAll: (params) => {
+    const convertedParams = strapi.utils.models.convertParams('user', params);
+
+    return strapi.plugins['users-permissions'].models.user
+      .find()
+      .where(convertedParams.where)
+      .sort(convertedParams.sort)
+      .skip(convertedParams.start)
+      .limit(convertedParams.limit)
+      .populate(_.keys(_.groupBy(_.reject(strapi.plugins['users-permissions'].models.user.associations, {autoPopulate: false}), 'alias')).join(' '));
+  },
+
+  /**
+   * Promise to fetch a/an user.
+   *
+   * @return {Promise}
+   */
+
+  fetch: (params) => {
+    return strapi.plugins['users-permissions'].models.user
+      .findOne(params)
+      .populate(_.keys(_.groupBy(_.reject(strapi.plugins['users-permissions'].models.user.associations, {autoPopulate: false}), 'alias')).join(' '));
+  },
+
+  /**
+   * Promise to add a/an user.
+   *
+   * @return {Promise}
+   */
+
+  add: async (values) => {
+    if (values.password) {
+      values.password = await strapi.plugins['users-permissions'].services.user.hashPassword(values);
+    }
+
+    const data = await strapi.plugins['users-permissions'].models.user.create(_.omit(values, _.keys(_.groupBy(strapi.plugins['users-permissions'].models.user.associations, 'alias'))));
```

Comment:
Above we are only accepting attribute which are defined in the model https://github.com/strapi/strapi/pull/346/files#diff-fe9fb26bf0fde2d06666a9154be1c050R34 and here we are only removing associations attributes. Why do we have two different strategies?

Also, we should use the `strapi.query` method to handle ORMs